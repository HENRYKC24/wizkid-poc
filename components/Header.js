import React, { useState } from "react";
import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";

import profilePic from "../assets/images/profile_pic.jpg";
import coin from "../assets/images/coin.png";
import back from "../assets/images/back.png";
import settings from "../assets/images/settings.png";

const Header = ({ navigation, showSettings }) => {
  const [activeTab, setActiveTab] = useState("spending");
  return (
    <View style={styles.container}>
      <View style={styles.imageCoin}>
        <TouchableOpacity>
          <Image source={profilePic} style={styles.profile} />
        </TouchableOpacity>
        <TouchableOpacity>
          <View style={styles.coinValue}>
            <Image source={coin} />
            <Text style={styles.amount}>80K</Text>
          </View>
        </TouchableOpacity>
      </View>
      <View style={styles.spendSave}>
        <TouchableOpacity onPress={() => setActiveTab("spending")}>
          <Text
            style={[
              styles.spend,
              activeTab === "spending" ? styles.active : "",
            ]}
          >
            Spending
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => setActiveTab("saving")}>
          <Text style={[activeTab === "saving" ? styles.active : ""]}>
            Saving
          </Text>
        </TouchableOpacity>
      </View>
      <View style={styles.backSettings}>
        <TouchableOpacity onPress={() => navigation.navigate("Welcome")}>
          <Image style={styles.back} source={back} />
        </TouchableOpacity>
        {showSettings ? (
          <TouchableOpacity>
            <Image style={styles.settings} source={settings} />
          </TouchableOpacity>
        ) : (
          <Text></Text>
        )}
      </View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "flex-end",
    position: "static",
    zIndex: 2,
    paddingBottom: 20,
  },
  active: {
    fontWeight: "700",
    textDecorationLine: "underline",
    textDecorationStyle: "solid",
    textDecorationColor: "#000",
    color: "#000",
  },
  profile: {
    width: 40,
    height: 40,
    borderRadius: 100,
  },
  imageCoin: {
    marginBottom: -20,
  },
  coinValue: {
    flexDirection: "row",
  },
  amount: {
    fontWeight: "800",
    color: "#888",
  },
  spendSave: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    gap: 20,
  },
  spend: {
    marginRight: 20,
  },
  backSettings: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "flex-end",
  },
  back: {
    marginBottom: 15,
  },
  settings: {
    marginLeft: 8,
  },
});
