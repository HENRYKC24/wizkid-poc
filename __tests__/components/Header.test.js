import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../components/Header';

describe('<Header />', () => {
  it('has 3 children', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree.children.length).toBe(3);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});