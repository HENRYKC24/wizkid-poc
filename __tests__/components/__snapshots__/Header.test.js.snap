// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 20,
      "position": "static",
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": -20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 100,
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
        />
        <Text
          style={
            Object {
              "color": "#888",
              "fontWeight": "800",
            }
          }
        >
          80K
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "gap": 20,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "marginRight": 20,
            },
            Object {
              "color": "#000",
              "fontWeight": "700",
              "textDecorationColor": "#000",
              "textDecorationLine": "underline",
              "textDecorationStyle": "solid",
            },
          ]
        }
      >
        Spending
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            "",
          ]
        }
      >
        Saving
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "marginBottom": 15,
          }
        }
      />
    </View>
    <Text />
  </View>
</View>
`;
